src/main/java/org/sagebionetworks/bridge/models/activities/StudyActivityEvent.java

```diff
@@ -42,55 +50,54 @@
     private DateTime createdOn;
     @Transient
     private int recordCount;
+    @Transient
+    private ActivityEventUpdateType updateType;
     
-    public StudyActivityEvent() {}
+    // Hibernate is happy with default (package) visibility constructor and setters, ensuring
+    // that objects can only be constructed in code through a builder.
     
-    public StudyActivityEvent(String eventId, DateTime timestamp) {
-        this.eventId = eventId;
-        this.timestamp = timestamp;
-        this.recordCount = 1;
-    }
+    StudyActivityEvent() {}
```

Comment:
What does Hibernate really need here? Remove more if possible.

Reply:
No it doesn't need any of it. When you put the annotations on the field, it can set them even without accessors. All removed.